### PR TITLE
fix/maid-manager: overhaul account refresh

### PR DESCRIPTION
**Pull request by madadam**

### CODE REVIEW

File: src/personas/maid_manager/mod.rs [hunk 1]
```diff
@@ -83,7 +94,8 @@ impl MaidManager {
                                    dst: Authority<XorName>,
                                    msg_id: MessageId)
                                    -> Result<(), InternalError> {
-        let res = self.get_account(&src, &dst).map(|account| account.info);
+        let res = self.get_account(&src, &dst)
+            .map(|account| account.balance());
```
Comment (afck) [inline src/personas/maid_manager/mod.rs]: Could that just be `.map(Account::balance)`?

File: src/personas/maid_manager/mod.rs [hunk 2]
```diff
@@ -471,59 +474,54 @@ impl MaidManager {
...
+        let msg_ids_to_delete: Vec<_> = self.request_cache
             .iter()
-            .filter(|&(_, entry)| accounts_to_delete.contains(&entry.src.name()))
-            .map(|(msg_id, _)| *msg_id)
-            .collect_vec();
+            .filter_map(|(msg_id, entry)| if accounts_to_delete.contains(&entry.src.name()) {
+                            Some(*msg_id)
+                        } else {
+                            None
+                        })
```
Comment (afck) [inline src/personas/maid_manager/mod.rs:503]: (I find `filter` and `map` was more concise, but that's a matter of taste.)

File: src/personas/maid_manager/tests.rs [hunk 3]
```diff
@@ -80,6 +81,9 @@ fn idata_basics() {
     assert_eq!(message.dst, client);
     assert_match!(message.response, Response::PutIData { res: Ok(()), .. });
 
+    // Simulate refresh from at least QUORUM nodes.
+    simulate_refresh(&mut node, &mut mm, msg_id, QUORUM);
```
Comment (afck) [inline src/personas/maid_manager/tests.rs:85]: That shouldn't be necessary anymore, now that we update the count on `PutSuccess`? (Also below.)

**Comment by madadam** [comment]
Wrong approach - closing.

**Pull request closed without merging**